README.md

```diff
@@ -139,162 +168,228 @@ Then you can run SwiftLint in the current directory with this command:
+by the build tool plugin. This is because it isn't possible to pass
+arguments to build tool plugins (e.g., passing the config file path).
 
-If your project structure doesn't work directly with the build tool 
+If your project structure doesn't work directly with the build tool
 plugin, please consider one of the following options:
 
-- To use a config file located outside the package/project directory, a config 
-  file may be added to that directory specifying a parent config path to the 
+- To use a config file located outside the package/project directory, a config
+  file may be added to that directory specifying a parent config path to the
   other config file, e.g., `parent_config: path/to/.swiftlint.yml`.
-- You can also consider the use of a Run Script Build Phase in place of the
-  build tool plugin.
+- You can also consider the use of a
+  [Run Script Build Phase](#xcode-run-script-build-phase) in place of the build
+  tool plugin.
 
-#### Xcode Projects
+### Xcode Run Script Build Phase
 
-You can integrate SwiftLint as an Xcode Build Tool Plugin if you're working
-with a project in Xcode.
+If the build tool plugin does not work for your project setup or when
+additional custom setup is required, SwiftLint can be added as a Run Script
+Build Phase. This is useful when a project setup relies on the `--config`
+SwiftLint option; or to lint all targets together in a single `swiftlint`
+invocation. File inclusions and exclusions can be configured in the
+[`.swiftlint.yml` configuration](#configuration).
 
-Add SwiftLint as a package dependency to your project without linking any of the
-products.
+To do this, add a custom script to a `Run Script` phase of the `Build Phases`
+of the primary app target, after the `Compile Sources` phase. Use the
+following script implementation:
 
-Select the target you want to add linting to and open the `Build Phases` inspector.
-Open `Run Build Tool Plugins` and select the `+` button.
-Select `SwiftLintBuildToolPlugin` from the list and add it to the project.
+```
+if command -v swiftlint >/dev/null 2>&1
+then
+    swiftlint
+else
+    echo "warning: `swiftlint` command not found - See https://github.com/realm/SwiftLint#installation for installation instructions."
+fi
+```
 
-![](https://raw.githubusercontent.com/realm/SwiftLint/main/assets/select-swiftlint-plugin.png)
+<br>
 
-For unattended use (e.g. on CI), you can disable the package and macro validation dialog by
+> [!TIP]
+> Uncheck `Based on dependency analysis` to run `swiftlint` on all incremental
+> builds, suppressing the unspecified outputs warning.
 
-* individually passing `-skipPackagePluginValidation` and `-skipMacroValidation` to `xcodebuild` or
-* globally setting `defaults write com.apple.dt.Xcode IDESkipPackagePluginFingerprintValidatation -bool YES` and `defaults write com.apple.dt.Xcode IDESkipMacroFingerprintValidation -bool YES` 
-  for that user.
+> [!IMPORTANT]
```

Comment:
I believe that can be a note instead. (allows the other important callouts to stand out more)
```suggestion
> [!NOTE]
```